### ISSUE TITLE
pytorch study 3주차

### ISSUE THREAD

**Issue by hyunsu-yang** [hyunsu-yang/PyTorch#2]
## 강의
- https://www.edwith.org/boostcourse-dl-pytorch/joinLectures/22155
## 목차 
- Lab-06 Softmax Classification
- Lab-07-1 Tips
- Lab-07-2 MNIST Introduction

**Comment by hyunsu-yang** [comment]
## Lab-06 Softmax Classification
### 핵심 키워드&요약
-  one hot encoding
- 소프트맥스(softmax) 함수
  - 소프트맥스 회귀에서는 비용 함수로 크로스 엔트로피 함수를 사용
  - 소프트맥스의 클래스의 개수가 2일 때 비용함수는 결국 로지스틱 회귀의 비용 함수와 동일 함
- 소프트맥스 비용함수 구하기의 여러 방법
  - 1. F.softmax() + torch.log() = F.log_softmax()
  - 2. F.log_softmax() + F.nll_loss() = F.cross_entropy()
  - F.cross_entropy는 비용 함수에 소프트맥스 함수까지 포함하고 있으므로 가설에서는
    소프트맥스 함수를 사용할 필요가 없음.

### 문제
- 소프트맥스 함수 계산을 데이터 정의시 one hot encoding 필용한 이유
- DataLoader 사용시 drop_last=True 설정하는 이유
-  과적합(Overfitting) 을 최소화 하려면
![image](https://user-images.githubusercontent.com/19383191/92423435-14015a00-f1bc-11ea-9baf-022d18ccd125.png)

**Comment by hyunsu-yang** [comment]
## Lab-06 Softmax Classification
#### 1. softmax와 sigmoid의 관계는?
-  로지스틱 회귀에서 시그모이드 함수는 예측값을 0과 1 사이의 값으로 만들고 두 클래스로 분류하여 합은 1.
-  소포트맥스 회귀에서는  세 클래스 이상  분류하여 합은 1
- H(X)=softmax(WX+B) 

#### 1. cross entropy와 binary cross entropy의 관계는?
- 소프트맥스 회귀에서는 비용 함수로 크로스 엔트로피 함수 사용.
- 로지스틱 회귀에서 배운 크로스 엔트로피 함수식과 달라보이지만, 본질적으로는 동일한 함수식
-  softmax에서 클래스의 개수가 2일 때 비용함수는 결국 로지스틱 회귀의 비용 함수와 동일 함
- 결국 2개로 분류하는지 3개이상으로 분류하는지의 차이일 뿐

## Lab-07-1 Tips
#### 1. learning rate가 너무 작은 경우 어떤 문제가 발생하는지?
- 학습 속도가 매우 느려지고 cost가 줄어들지 않음.

#### 1. standardization 효과는 무엇인가?
- standardization (정규 분포화)
```
기존 변수에 범위를 정규 분포로 변환한다.
식) (X - X의 평균값) / (X의 표준편차)
데이터의 최소값, 최대값을 모를 경우 사용
표준화를 해주면 정규화처럼 특성값의 범위가 0과 1의 범위로 균일하게 바뀌지는 않음.
```
- Min-Max Normalization
 ```
데이터를 일반적으로 0~1 사이의 값으로 변환시켜준다.
식) (X - X의 최소값) / (X의 최대값 - X의 최소값)
데이터의 최소값, 최대값을 알 경우 사용
```
## Lab-07-2 MNIST Introduction
#### 1. epoch, batch size, iteration 각각이 무엇을 의미하는지?
- epoch :  전체 훈련 데이터가 학습에 한 번 사용된 주기
- batch:   전체 데이터를 더 작은 단위로 나누어서 해당 단위로 학습하는 개념
- iteration : 이 배치를 몇 번 사용했는지
![image](https://user-images.githubusercontent.com/19383191/92422473-88d29500-f1b8-11ea-8943-3e2023dc073c.png)


#### 1. batch size가 작을 때와 클 때 어떤 차이가 있는지?
- 너무 작게 되면 제대로 학습이 되지 않을 듯 한데..iteration 돌 때 마다 학습 데이터가 적고 편차가 심해 질 수 있어서 ?
- 반대로 크게 되면 batch 없이 그냥 epoch 단위로 학습하는 것과 차이가 없을 것 같음.
